### DISCUSSION TITLE
Middle click scroll

### DISCUSSION THREAD

**Discussion by raspin0** [linuxdeepin/developer-center#3461] [Features Request & Ideas | 特性请求 & 头脑风暴]
Currently middle click paste is more annoying than useful. I miss this functionality from Windows where you could use middle click to scroll through application contents and even Firefox added such option to enable for Linux users. I heard Wayland provides more flexibility on that front. Would it be possible to bring that to deepin? 

It doesn't have to be hard coded globally, but even as an option in DTK for parts of applications. It would be cool to scroll the System Monitor processes, documents and websites using this functionality.

**Comment by BLumia** [comment]
> Currently middle click paste is more annoying than useful.

It's a default setting of x11/xorg server in general actually, not related to DTK. see also: https://unix.stackexchange.com/questions/24330/how-can-i-turn-off-middle-mouse-button-paste-functionality-in-all-programs/472464#472464

> I miss this functionality from Windows where you could use middle click to scroll through application contents

I did a test under Windows 11 and MMB click+drag won't trigger scrolling in its File Explorer, Process Manager, Video and Photo Viewer app (I'm using non-English locale so maybe that's not the actual name). Web browsers usually have this feature (on both Windows and Linux) so maybe you are getting used to this feature in a browser and might think it's a widely-supported feature?

**Reply by raspin0** [reply]
> It's a default setting of x11/xorg server in general actually, not related to DTK. see also: https://unix.stackexchange.com/questions/24330/how-can-i-turn-off-middle-mouse-button-paste-functionality-in-all-programs/472464#472464

Yes on X11 apparently it's hard coded, but newest versions of wayland KDE allow to disable this feature in the settings. I tried a couple of workarounds for disabling middle click paste on X11, but they didn't seem to be very reliable.

> I did a test under Windows 11 and MMB click+drag won't trigger scrolling in its File Explorer, Process Manager, Video and Photo Viewer app (I'm using non-English locale so maybe that's not the actual name). Web browsers usually have this feature (on both Windows and Linux) so maybe you are getting used to this feature in a browser and might think it's a widely-supported feature?

Chromium based browsers doesn't seem to support this on Linux, at least not by default. On Windows the functionality seems to be present everywhere it's useful - Browsers, Electron based apps (ex. discord), Word processors (Wordpad, MS Word), PDF viewers etc.. 

My point is that deepin team should think about the future of Middle Click. I was referencing to DTK, because I believe it could be implemented in the toolkit and be used in parts of applications where it's actually useful. Personally I'm a huge fan of Middle Click Scroll, because it's a faster and more convenient alternative to scrollbars that these days are getting ever so smaller and often simply disappear.

**Reply by BLumia** [reply]
cc @meiyixiang

**Reply by raspin0** [reply]
@BLumia any update on this?

**Reply by BLumia** [reply]
Sorry, no related update currently. We are working on Wayland support in v23 though.

**Reply by raspin0** [reply]
> Sorry, no related update currently. We are working on Wayland support in v23 though.

Hello, 3 years later 😅. Since you have starting developing your own wayland compositor, would system-wide autoscroll be possible on treeland? Don't know if you have this on your to-do list, but since a lot of issues have been improved since then I was wondering if this feature could get some attention. Still a huge missing feature in my opinion.

**Reply by BLumia** [reply]
Treeland support is still not considered as stable so I guess treeland team will still focus on usability than features atm...
